### ISSUE TITLE
[Bug] jedis-4.x-plugin incorrectly took effect in jedis 2.x version.

### ISSUE THREAD

**Issue by xzyJavaX** [apache/skywalking#10679]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

jedis-4.x-plugin has incorrectly taken effect in jedis 2.5.2, 2.6.3 versions, but jedis 2.7.3, 2.8.2, 2.9.0 versions have no impact.
The witness method for jedis-4.x-plugin selection is 
`new WitnessMethod("redis.clients.jedis.Pipeline", ElementMatchers.named("persist").and(ElementMatchers.takesArguments(1)))`.
In jedis 2.5.2, 2.6.3 versions，when loading the jedis plugin，`org.apache.skywalking.apm.agent.core.plugin.WitnessFinder#exist(WitnessMethod , ClassLoader)` returns true.

### What you expected to happen

No response

### How to reproduce

No response

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Are you going to fix this?

**Comment by lujiajing1126** [comment]
We are not clear about what actually happened in these cases. But it is abnormal.

There is no significant differences of `redis.clients.jedis.Pipeline` at source code level between versions `2.5.2, 2.6.3` and `2.7.3, 2.8.2, 2.9.0`. But when we were debugging the [WitnessFinder](https://github.com/apache/skywalking-java/blob/24858574474e0656e13c592a2c38d83b0637c7f5/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/WitnessFinder.java#L74), we've observed over 300 hundreds method tokens for `redis.clients.jedis.Pipeline` from `2.5.2, 2.6.3` but only ~10 method tokens for `redis.clients.jedis.Pipeline` from `2.7.3, 2.8.2, 2.9.0`.

One possible reason is that the resolved `redis.clients.jedis.Pipeline` has already been transformed by another plugin and then it gave wrong `declaredMethods`. We will further investigate this issue.

**Comment by wu-sheng** [comment]
I think unless you have another SkyWalking kernel running, it is impossible it has been instrumented.

**Comment by xzyJavaX** [comment]
jedis 2.5.2
![920C58A5-CF08-4AF4-8D49-9EEE60926B6B](https://user-images.githubusercontent.com/49760141/231731807-e1e771f2-ebda-4461-912a-05682aef2977.png)
![29e076c6-8b50-40c0-9cf9-be2471065700](https://user-images.githubusercontent.com/49760141/231731842-2df8947c-b44b-4537-9c3a-b427ed2ff35a.png)
![0438d58f-9f3d-4b62-86fb-90d7dc8a1369](https://user-images.githubusercontent.com/49760141/231731854-8bee3881-9dce-4483-b3d6-079c192147cb.png)

**Comment by xzyJavaX** [comment]
There should be no such methods in the `redis.clients.jedis.Pipeline`. These methods are enhanced in `redis.clients.jedis.PipelineBase`, Pipeline inherits PipelineBase.

**Comment by lujiajing1126** [comment]
> Are these different between versions? I am not following.

Finally, I confirm the issue is from the upstream. I've made a minimal example to reproduce this issue,

https://github.com/raphw/byte-buddy/issues/1429#issuecomment-1507111558

Let's see what the author would reply. It would be a resolution bug or miss usage of byte-buddy API.

**Comment by wu-sheng** [comment]
This seems a very strange bug. Let's see.

**Comment by lujiajing1126** [comment]
> This seems like a very strange bug. Let's see.

As is discussed in the JDK community, https://bugs.openjdk.org/browse/JDK-6815786 (lasting for 11 years), this is an expected behavior of Java compiler. Bridges methods will be generated for public methods in the non-public superclasses. Specifically for Jedis, https://github.com/redis/jedis/commit/b248532bf61594ed1246b42c08a9aafdafc0dd6f this commit added `public` modifier to the `MultiKeyPipelineBase` and `PipelineBase`. This causes observed differences in this issue.

In conclusion, for SkyWalking scenario, we should exclude `synthetic` methods in the `WitnessFinder`. I will raise a PR to resolve this issue.
